winpr/libwinpr/sspi/Kerberos/kerberos.c

```diff
@@ -1351,72 +1408,179 @@ static SECURITY_STATUS SEC_ENTRY kerberos_DeleteSecurityContext(PCtxtHandle phCo
+	 * since Windows 2000 SP2 and still remains in Windows 7 and Windows 2008 R2.
+	 *  For Windows Server 2012, the default value of the MaxTokenSize registry
+	 *  entry is 48,000 bytes.*/
+	ContextSizes->cbMaxToken = KERBEROS_SecPkgInfoA.cbMaxToken;
+	ContextSizes->cbMaxSignature = 0;
+	ContextSizes->cbBlockSize = 1;
+	ContextSizes->cbSecurityTrailer = 0;
+
+	key = get_key(&context->keyset);
+
+	if (context->flags & SSPI_GSS_C_CONF_FLAG)
+	{
+		krb5_error_code rv = krb_log_exec(krb5glue_crypto_length, credentials->ctx, key,
+		                                  KRB5_CRYPTO_TYPE_HEADER, &header);
+		if (rv)
+			return krb5_error_to_SECURITY_STATUS(rv);
+
+		rv = krb_log_exec(krb5glue_crypto_length, credentials->ctx, key, KRB5_CRYPTO_TYPE_PADDING,
+		                  &pad);
+		if (rv)
+			return krb5_error_to_SECURITY_STATUS(rv);
+
+		rv = krb_log_exec(krb5glue_crypto_length, credentials->ctx, key, KRB5_CRYPTO_TYPE_TRAILER,
+		                  &trailer);
+		if (rv)
+			return krb5_error_to_SECURITY_STATUS(rv);
+
+		/* GSS header (= 16 bytes) + encrypted header = 32 bytes */
+		ContextSizes->cbSecurityTrailer = header + pad + trailer + 32;
+	}
+
+	if (context->flags & SSPI_GSS_C_INTEG_FLAG)
+	{
+		krb5_error_code rv = krb_log_exec(krb5glue_crypto_length, credentials->ctx, key,
+		                                  KRB5_CRYPTO_TYPE_CHECKSUM, &ContextSizes->cbMaxSignature);
+		if (rv)
+			return krb5_error_to_SECURITY_STATUS(rv);
+
+		ContextSizes->cbMaxSignature += 16;
+	}
+
+	return SEC_E_OK;
+}
+
+static SECURITY_STATUS kerberos_ATTR_TICKET_LOGON(KRB_CONTEXT* context,
+                                                  KRB_CREDENTIALS* credentials,
+                                                  KERB_TICKET_LOGON* ticketLogon)
+{
+	krb5_creds matchCred = { 0 };
+	krb5_auth_context authContext = NULL;
+	int getCredsFlags = KRB5_GC_CACHED;
+	BOOL firstRun = TRUE;
+	krb5_creds* hostCred = NULL;
+	SECURITY_STATUS ret = SEC_E_INSUFFICIENT_MEMORY;
+	int rv = krb_log_exec(krb5_sname_to_principal, credentials->ctx, context->targetHost, "HOST",
+	                      KRB5_NT_SRV_HST, &matchCred.server);
+	if (rv)
+		goto out;
+
+	rv = krb_log_exec(krb5_cc_get_principal, credentials->ctx, credentials->ccache,
+	                  &matchCred.client);
+	if (rv)
+		goto out;
+
+	/* try from the cache first, and then do a new request */
+again:
+	rv = krb_log_exec(krb5_get_credentials, credentials->ctx, getCredsFlags, credentials->ccache,
+	                  &matchCred, &hostCred);
+	switch (rv)
+	{
+		case 0:
+			break;
+		case KRB5_CC_NOTFOUND:
+			getCredsFlags = 0;
+			if (firstRun)
+			{
+				firstRun = FALSE;
+				goto again;
+			}
+			WINPR_FALLTHROUGH
+		default:
+			WLog_ERR(TAG, "krb5_get_credentials(hostCreds), rv=%d", rv);
+			goto out;
+	}
+
+	if (krb_log_exec(krb5_auth_con_init, credentials->ctx, &authContext))
+		goto out;
+
+	krb5_data derOut = { 0 };
+	if (krb_log_exec(krb5_fwd_tgt_creds, credentials->ctx, authContext, context->targetHost,
```

Comment:
Re: [comment](https://github.com/FreeRDP/FreeRDP/pull/10549#issuecomment-2333421875)

When I simply ignore this KRB5KDC_ERR_BADOPTION  error and continue instead of jumping to `out`, a connection is established and (despite some wild keyboard mapping) I can "work".